build/PackageArchive.targets

```diff
@@ -4,48 +4,68 @@
+
+      <_MetapackageReferenceType Include="AppMetapackage;AllMetapackage"/>
+    </ItemGroup>
 
     <!-- Copy the archive template -->
     <Copy SourceFiles="$(_TemplatesDirectory)Archive\Archive.csproj" DestinationFiles="$(_WorkRoot)Archive.csproj" />
 
-    <!-- Copy the archive template -->
+    <!-- Add LZMA references -->
     <RepoTasks.AddArchiveReferences
       ReferencePackagePath="$(_WorkRoot)Archive.csproj"
+      TargetFramework="$(ArchiveTargetFramework)"
       BuildArtifacts="@(ArtifactInfo)"
       PackageArtifacts="@(PackageArtifact)"
       ExternalDependencies="@(ExternalDependency)"
       MetapackageVersion="$(PackageVersion)" />
 
-    <PropertyGroup>
-      <FallbackStagingDir>$(_WorkRoot)obj\$(LzmaOutputPackageName)</FallbackStagingDir>
-      <FallbackOutputDir>$(ArtifactsDir)lzma\</FallbackOutputDir>
-      <FallbackOutputPath>$(FallbackOutputDir)$(LzmaOutputPackageName).lzma</FallbackOutputPath>
-      <GeneratedFallbackRestoreSourcesPropsPath>$(_WorkRoot)restoresources.$(LzmaOutputPackageName).props</GeneratedFallbackRestoreSourcesPropsPath>
-    </PropertyGroup>
-
-    <ItemGroup>
-      <_FallbackArchiveRestoreSources Include="$(RestoreSources)" />
-      <_FallbackArchiveRestoreSources Include="$(BuildDir)" Condition="Exists($(BuildDir))" />
-      <_FallbackArchiveRestoreSources Include="$(MetapackageRestoreSource)" Condition="Exists($(MetapackageRestoreSource))" />
-    </ItemGroup>
+    <!-- Add metapackage references to project -->
+    <RepoTasks.AddMetapackageReferences
+      ReferencePackagePath="$(_WorkRoot)Archive.csproj"
+      MetapackageReferenceType="%(_MetapackageReferenceType.Identity)"
+      LockToExactVersions="false"
+      BuildArtifacts="@(ArtifactInfo)"
+      PackageArtifacts="@(PackageArtifact)"
+      ExternalDependencies="@(ExternalDependency)" />
 
     <RepoTasks.GenerateRestoreSourcesPropsFile
       Sources="@(_FallbackArchiveRestoreSources)"
       OutputPath="$(GeneratedFallbackRestoreSourcesPropsPath)" />
 
-    <!-- Create the Staging Dir -->
-    <MakeDir Directories="$(FallbackStagingDir);$(FallbackOutputDir)" />
-
     <!-- Restore the target project -->
     <MSBuild
       Projects="$(_WorkRoot)Archive.csproj"
       Targets="Restore"
-      Properties="RestorePackagesPath=$(FallbackStagingDir);RuntimeFrameworkVersion=$(MicrosoftNETCoreApp21PackageVersion);DotNetRestoreSourcePropsPath=$(GeneratedFallbackRestoreSourcesPropsPath);DotNetBuildOffline=true;AspNetUniverseBuildOffline=true" />
+      Properties="ArchiveTargetFramework=$(ArchiveTargetFramework);RestorePackagesPath=$(FallbackStagingDir);RuntimeFrameworkVersion=$(MicrosoftNETCoreApp21PackageVersion);DotNetRestoreSourcePropsPath=$(GeneratedFallbackRestoreSourcesPropsPath);DotNetBuildOffline=true;AspNetUniverseBuildOffline=true" />
+  </Target>
+
+  <Target Name="BuildFallbackArchive" DependsOnTargets="ResolveRepoInfo">
+    <ItemGroup>
+      <_ArchiveTargetFrameworks Include="netstandard2.0;netcoreapp2.1"/>
```

Comment:
We want to be able to restore for netstandard2.0 for class libraries and netcoreapp2.1 for apps.